src/backend/cuda/kernel/ireduce.hpp

```diff
@@ -36,13 +55,16 @@ namespace kernel
         __host__ __device__ MinMaxOp(T val, uint idx) :
             m_val(val), m_idx(idx)
         {
+            if (isNan(val)) {
+                m_val = Binary<T, op>::init();
+            }
         }
 
         __host__ __device__ void operator()(T val, uint idx)
         {
-            if (cabs(val) < cabs(m_val) ||
-                (cabs(val) == cabs(m_val) &&
-                 idx > m_idx)) {
```

Comment:
I think this condition was incorrectly removed. See comment below.

Reply:
Addressed this concern in my comment in the related change too (the max counterpart), please let me know what you think in that thread.